crates/ruff/src/rules/tryceratops/rules/pointless_raise.rs

```diff
@@ -0,0 +1,58 @@
+
+
+use rustpython_parser::ast::{self, Excepthandler, StmtKind, ExcepthandlerKind};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+
+use crate::checkers::ast::Checker;
+
+
+/// ## What it does
+/// Checks for uses of `raise` directly after a `rescue`.
+///
+/// ## Why is this bad?
+/// Catching an error just to reraise it is pointless. Instead, remove error-handling and let the error propogate naturally
+///
+/// ## Example
+/// ```python
+///
+/// def foo():
+///     try:
+///         bar()
+///     except NotImplementedError:
+//          raise
+/// ```
+///
+/// Use instead:
+/// ```python
+///
+/// def foo():
+///     bar()
+/// ```
+#[violation]
+pub struct PointlessRaise;
+
+impl Violation for PointlessRaise {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Remove this exception handler as the error is immediately re-raised")
+    }
+}
+
+/// TRY302
+pub(crate) fn pointless_raise(checker: &mut Checker, handlers: &[Excepthandler]) {
+    for handler in handlers {
+        let ExcepthandlerKind::ExceptHandler(handler) = &handler.node;
+        let body = &handler.body;
+
+        if let Some(stmt) = body.first() {
+            if let StmtKind::Raise(ast::StmtRaise { exc: None, .. }) = &stmt.node {
```

Comment:
If you take a look at the logic in [tryceratops](https://github.com/guilatrova/tryceratops/commit/f7a56dd337acda060aa643c80f20b2e780ad2fa2#diff-a2fd5eddbabe84a3b436a6be3501fa89c8d23023c77771f3d7e84b9e19ce9c05R158), they do a couple nice things:

1. They check if _all_ handlers are useless. I think this is correct, since you can use a "useless" handler to omit exceptions from subsequent handlers.
2. They check that the body is _only_ a raise, and that the argument is either the same as the "cause" (`except ... as e: raise e`) or implicit (`except ... as e: raise`). I think we should check those too.

Might also make sense to take a look at their test suite to ensure compatibility.

Reply:
point 1 and the latter half of point 2 makes sense, I handle the latter style but not the former, shall fix that.

About the body only being a raise, what's the reasoning for that? Surely
```py
except:
    raise
    print("something went wrong!")
```
would warrant both this error and some sort of dead-code error?

Although make sense if the aim is preserving exact behaviour

Reply:
Ah yeah, I think you're right. It's probably fine to retain the "first statement is a raise" behavior you have here.

Reply:
:+1: addressed the comments